sql/core/src/test/scala/org/apache/spark/sql/execution/columnar/PartitionBatchPruningSuite.scala

```diff
@@ -95,6 +111,17 @@ class PartitionBatchPruningSuite
   checkBatchPruning("SELECT key FROM pruningData WHERE 11 >= key", 1, 2)(1 to 11)
   checkBatchPruning("SELECT key FROM pruningData WHERE 88 < key", 1, 2)(89 to 100)
   checkBatchPruning("SELECT key FROM pruningData WHERE 89 <= key", 1, 2)(89 to 100)
+  // Do not filter on array type
+  checkBatchPruning("SELECT _1 FROM pruningArrayData WHERE _1 = array(1)", 5, 10)(Seq(Array(1)))
+  checkBatchPruning("SELECT _1 FROM pruningArrayData WHERE _1 <= array(1)", 5, 10)(Seq(Array(1)))
+  checkBatchPruning("SELECT _1 FROM pruningArrayData WHERE _1 >= array(1)", 5, 10)(
+    testArrayData.map(_._1))
+  // Do not filter on binary type
+  checkBatchPruning(
+    title = "SELECT _1 FROM pruningBinaryData WHERE _1 == 0x01 (binary literal)",
+    actual = spark.table("pruningBinaryData").filter($"_1".equalTo(Array[Byte](1.toByte))),
```

Comment:
The problem here is, there seems no SQL liternal. So, I had to use DSL

Reply:
not very elegant, but we can do `binary(chr(5))` in order to get a binary literal

Reply:
that seems making a cast ..

Reply:
Oops right.